src/main/scala/org/coroutines/Coroutine.scala

```diff
@@ -124,6 +134,14 @@ object Coroutine {
       } else throw new CoroutineStoppedException
     }
 
+    /**
+      * Usage:
```

Comment:
Comment that if `pull` returns `true`, indicating that there are more elements to traverse, then the coroutine has suspended by yielding a value.